pkg/policy/rule.go

```diff
@@ -27,6 +29,19 @@ import (
 
 type rule struct {
 	api.Rule
+
+	// mutex protects everything below it (i.e., not the rule itself).
+	mutex lock.RWMutex
+
+	// localRuleConsumers is the set of the numeric identifiers which this rule
+	// selects which are node-local (e.g., Endpoint).
+	localRuleConsumers map[uint16]*identity.Identity
```

Comment:
I wonder if I could check if the pointers for the Identity are the same when looking up the identifier - it would ensure that if the endpoint ID is reused, but maps to a different identity, that we wouldn't incorrectly state that rule selects vs. it not selecting.